datafusion/src/physical_plan/repartition.rs

```diff
@@ -55,11 +56,14 @@ pub struct RepartitionExec {
     /// Partitioning scheme to use
     partitioning: Partitioning,
     /// Channels for sending batches from input partitions to output partitions.
-    /// Key is the partition number
-    channels: Arc<
-        Mutex<
+    /// Key is the partition number.
+    ///
+    /// Stored alongside is an abort marker that will kill the background job once it's no longer needed.
+    channels_and_abort_helper: Arc<
```

Comment:
What would you think about using a named struct (that can have additional documentation on it) rather than

```rust
(HashMap<usize, (UnboundedSender<MaybeBatch>, UnboundedReceiver<MaybeBatch>)>,
            Arc<AbortOnDrop>,)
```

to 

```rust
struct ChannelsAndAbortHelper {
  channels: HashMap<usize, (UnboundedSender<MaybeBatch>, UnboundedReceiver<MaybeBatch>)>,
  abort: Arc<AbortOnDrop>
}
```

And then you could rename `channels_and_abort_helper`  to `state` or something shorter and have the inner field name add semantic value

We can do that as a follow on refactor, but I wanted to get your opinion